dali/base/dadiags.cpp

```diff
@@ -122,7 +123,7 @@ class CDaliDiagnosticsServer: public IDaliServer, public Thread
                     mb.append(getReceiveQueueDetails(buf).str());
                 }
                 else if (0 == stricmp(id, "locks")) {
-                    mb.append(querySDS().getLocks(buf).str());
+                    querySDS().getLocks(mb);
```

Comment:
Unless I'm misreading, a "locks" request from dalidiag will [still] go via getDaliDiagnosticValue(const char *name, StringBuffer &ret).
Which is the default for anything that isn't specially handled by dalidiag, i.e. a generic command fulfiller.
It is expecting a string back... I don't see any formatting client side (in dalidiag.cpp).
Which is true of most of the other info requests, they are formatted server-side.
They _could_ serialize and format client side, but at the moment, afaics a "locks" would fall over.